### PR TITLE
Support for Revit 2023 added

### LINKED ISSUE #1249: Add support for Revit 2023
#### Description:

As in the title, support for Revit 2023 needed.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #1249

### Test files

Standard test procedures are the main test content plus a dedicated script [here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FRevit%5FToolkit%2F%231250%2DSupportForRevit2023&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4).

Versions to test with most care:
- one version between 2018 and 2020 (to double check if nothing breaks)
- 2021 - this is a bridge version that suffers most by the lack of consistency in versioning of Revit API (see additional comments)
- 2023 - new one

Key focus areas for testing:
- handling of parameters (push new parameter definitions, update element parameters incl. parameters with units etc., pulling elements with parameters)
- pulling structural bars and panels based on the analytical model

### Additional comments

Three main reasons for meaningful code changes in this PR, beyond simply adding the new version:
- `ParameterType` enum is gone, replaced with `ForgeTypeId` (which now encompasses `UnitType`, `DisplayUnitType` and `ParameterType`)
- renaming `GetSpecTypeId` method to `GetDataType`
- Existence of `SpecTypeId` class in 2021 API _without_ nested `Boolean`, `Int`, `String` etc. classes added in 2022, which means that my concept of mimicking that class for older versions exercised [here](https://github.com/BHoM/Revit_Toolkit/blob/main/Revit_Core_Engine/Objects/SpecTypeId.cs) does not work for Revit 2021
- Autodesk **removed** some structural analytical model classes from the API without prior announcement ❤️

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check installer

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check required

File: Revit_Core_Adapter/Revit_Core_Adapter.csproj [hunk 1]
```diff
@@ -246,6 +268,14 @@
       <HintPath>..\libs\2022\RevitAPIUI.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="RevitAPI" Condition="'$(Configuration)'=='Debug2023' Or '$(Configuration)'=='Release2023'">
```
[the remark this answers is not in the document]
Comment (FraserGreenroyd) [reply Revit_Core_Adapter/Revit_Core_Adapter.csproj:271]: It's shorter, but not explicit, and in these purposes, explicit is preferred 😄

[the remark this answers is not in the document]
Comment (pawelbaran) [reply Revit_Core_Adapter/Revit_Core_Adapter.csproj:271]: I could lean towards @vietle-bh's suggestion, but there is so many Revit .csprojs that I would blanket cover clean them up one day. Now just doing the minimal work required to make it work 😉

File: Revit_Core_Adapter/Listener/Files/BHoM_2023.Addin [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
```
[the remark this answers is not in the document]
Comment (pawelbaran) [reply Revit_Core_Adapter/Listener/Files/BHoM_2023.Addin:1]: Fair point @vietle-bh, thanks for spotting! Should be fixed now.

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Review by vietle-bh — APPROVED** [review]
The best thing that will ever happen to Revit 2023 👍

**Review by michal-pekacki — APPROVED** [review]
Tested on Revit 2018 and 2021 without issues! 👍

**Comment by pawelbaran** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check dataset-compliance
@BHoMBot check ready-to-merge

**Pull request merged**